creation/interfaces-and-networks.md

```diff
@@ -492,64 +492,52 @@ the interface should be configured as follows.
+#### masquerade - IPv4 and IPv6 dual-stack support
+
+It is currently experimental, but `masquerade` mode can be used in IPv4 and IPv6
+dual-stack clusters to provide a VM with an IP connectivity over both protocols.
 
 As with the IPv4 `masquerade` mode, the VM can be contacted using the pod's IP
-address - which will be an IPv6 one. Outgoing traffic is also "NAT'ed" to the
-pod's IPv6 address.
+address - which will be in this case two IP addresses, one IPv4 and one
+IPv6. Outgoing traffic is also "NAT'ed" to the pod's respective IP address
+from the given family.
 
-Unlike in IPv4, the configuration of the IP address, default route, and DNS is
+Unlike in IPv4, the configuration of the IPv6 address and the default route is
 not automatic; it should be configured via cloud init, as shown below:
 
-<pre>
-    kind: VM
-    metadata:
-      namespace: <i>your-favorite-namespace</i>
-    spec:
-      domain:
-        devices:
-          disks:
-            - disk:
-              bus: virtio
-              name: cloudinitdisk
-          interfaces:
-            - name: red
-              masquerade: {} # connect using masquerade mode
-              ports:
-                - port: 80 # allow incoming traffic on port 80 to get into the virtual machine
-      networks:
-      - name: red
-        pod: {}
-      volumes:
-      - cloudInitNoCloud:
-          networkData: |
-            version: 2
-            ethernets:
-              eth0:
-                addresses: [ fd10:0:2::2/120 ]
-                gateway6: fd10:0:2::1
-                nameservers:
-                  addresses: [ <i>dns_server_ip</i> ]
-                  search:
-                    - <i>your-favorite-namespace</i>.svc.cluster.local
-                    - svc.cluster.local
-                    - cluster.local
-          userData: |-
-            #!/bin/bash
-            echo "fedora" |passwd fedora --stdin
-</pre>
-
-> **Note:** The IP address for the VM and default gateway **must** be the ones
-> shown above.
-
-> **Note:** DNS must also be manually configured. To know which DNS server to
-> use, run the following query:
-> `kubectl get service -n kube-system kube-dns -o=custom-columns=IP:.spec.clusterIP`
+```yaml
+kind: VM
+spec:
+  domain:
+    devices:
+      disks:
+        - disk:
+          bus: virtio
+          name: cloudinitdisk
+      interfaces:
+        - name: red
+          masquerade: {} # connect using masquerade mode
+          ports:
+            - port: 80 # allow incoming traffic on port 80 to get into the virtual machine
+  networks:
+  - name: red
+    pod: {}
+  volumes:
+  - cloudInitNoCloud:
+      networkData: |
+        version: 2
+        ethernets:
+          eth0:
+            dhcp4: true
+            addresses: [ fd10:0:2::2/120 ]
+            gateway6: fd10:0:2::1
+      userData: |-
+        #!/bin/bash
+        echo "fedora" |passwd fedora --stdin
+```
 
-> **Note:** The DNS search domain namespace must match with the namespace where
```

Comment:
I don't think this is a fitting place for it. The user-guide is just for the end-user. DNS implementation details should probably go into kubevirt/kubevirt/docs. We should document the minimal example that would work, that's what the user hopefully looks for.

Reply:
In this current configuration, is DNS configuration  transparent? 
If so, is  there a scenario that it's not?

Reply:
DNS is configured in the same manner it is on IPv4 single-stack. It is obtained using DHCP client from the IPv4 DNS server.

AFAIK it should be always obtained. As long as it works with IPv4, it would work with dual-stack.

Reply:
> I don't think this is a fitting place for it. The user-guide is just for the end-user. DNS implementation details should probably go into kubevirt/kubevirt/docs. We should document the minimal example that would work, that's what the user hopefully looks for.

+1 on putting anything elaborating on how DNS works exclusively on the dev guide.

Reply:
ok then if that's the case then I'm good with this